smooth_scrolling_chapter

IlyaDevN opened the pull request.

IlyaDevN committed:
refactor: Add page scroll to separate sections

Nover98 commented on src/js/modules/scrollToAnchor.js:
```diff
@@ -0,0 +1,22 @@
+import { closeMenu } from './mobileMenu.js';
```
move all the code to mobileMenu file it is still all about menu

Nover98 replied:
and rename file as just "menu"

IlyaDevN replied:
Done

Nover98 commented on src/js/modules/scrollToAnchor.js:
```diff
@@ -0,0 +1,22 @@
+import { closeMenu } from './mobileMenu.js';
+
+const anchors = Array.from(document.querySelectorAll("a[href*='#']"));
```
why do you need Array.from ?

IlyaDevN replied:
Done

Nover98 commented on src/js/modules/scrollToAnchor.js:
```diff
@@ -0,0 +1,22 @@
+import { closeMenu } from './mobileMenu.js';
+
+const anchors = Array.from(document.querySelectorAll("a[href*='#']"));
```
all your menu links have class `nav__link`, use it in querySelectorAll

IlyaDevN replied:
There are nav__link-footer in footer section and they have other styles. So I need already used selector.

Nover98 replied:
```js
headerMenu.querySelectorAll(".nav__link");
```

IlyaDevN replied:
I don't understand you. Why only headerMenu? I have menu links not only in header but also in footer. 
![image](https://github.com/IlyaDevN/Horse-club/assets/106959524/443ae5f2-c411-409e-a849-8d1241d9f0b5)
![image](https://github.com/IlyaDevN/Horse-club/assets/106959524/70ca2aa4-f7b8-4f41-863c-73f1d5f875f1)

Nover98 replied:
I didn't pay attention that footer link lead to this section as well. Then 
```js
document.querySelectorAll(".nav__link-footer");
```
and join all the links in one collection

I don't like the idea to find all the anchors with such a selector `a[href*='#']`
To control more what you do and avoid all the possible bugs, just find exactly the elements you need.

IlyaDevN replied:
Done

Nover98 commented on src/js/modules/scrollToAnchor.js:
```diff
@@ -0,0 +1,22 @@
+})
+
+function closeMobileMenu() {
+	const menuOverlay = document.body.querySelector(".header__menu-overlay");
```
when you move code, use existing variable

IlyaDevN replied:
It doesn't need to be done after moving all this code into menu.js

Nover98 commented on src/js/modules/scrollToAnchor.js:
```diff
@@ -0,0 +1,22 @@
+		event.preventDefault();
+		closeMobileMenu();
+		const blockID = anchor.getAttribute("href");
+		document.querySelector(`${blockID}`).scrollIntoView({
```
getElementById workds faster, you have id in href

IlyaDevN replied:
Done

Nover98 replied:
`blockID` - is already string in variable, you don't need this `${blockID}`, just use `getElementById(blockID)`

IlyaDevN replied:
Done

IlyaDevN committed:
refactor: Correct page scroll to separate sections

IlyaDevN committed:
refactor: Change branch for deploy on Github

IlyaDevN committed:
refactor: Move scrollToAnchor into mobileMenu.js

IlyaDevN committed:
refactor: Rename mobileMenu.js to menu.js

IlyaDevN committed:
refactor: Replace querySelector with getEl...ById

IlyaDevN committed:
refactor: Replace `${blockID}` with blockID

IlyaDevN committed:
refactor: Replace a[href*='#'] with other selec..

Replace a[href*='#'] with ".nav__link", ".nav__link-footer" and unite
them together.
